tests/helpers/volumes.py

```diff
@@ -163,3 +173,54 @@ def create_vvds_on_volume( ansible_zos_module, volume):
         if vls_res.get("rc") == 0:
             return True
     return False
+
+
+def get_volume_and_unit(ansible_zos_module, path):
+    """Get an array of available volumes, and it's unit"""
+    # Using the command d u,dasd,online to fill an array of available volumes with the priority
+    # of of actives (A) and storage (STRG) first then online (O) and storage and if is needed, the
+    # private ones but actives then to get a flag if is available or not every volumes
+    # is a instance of a class to manage the use.
+    hosts = ansible_zos_module
+    list_volumes = []
+    all_volumes_list = []
+    priv_online = []
+    flag = False
+    iteration = 5
+    # The first run of the command d u,dasd,online,,n in the system can conclude with empty data
+    # to ensure get volumes is why require not more 5 runs and lastly one second of wait.
+    while not flag and iteration > 0:
+        all_volumes = hosts.all.zos_operator(cmd="d u,dasd,online,,65536")
+        time.sleep(1)
+        if all_volumes is not None:
+            for volume in all_volumes.contacted.values():
+                temp = volume.get('content')
+                if temp is not None:
+                    all_volumes_list += temp
+            flag = True if len(all_volumes_list) > 5 else False
+        iteration -= 1
+    # Check if the volume is of storage and is active on prefer but also online as a correct option
+    for info in all_volumes_list:
+        if "ACTIVATED" in info or "-D U," in info or "UNIT" in info:
+            continue
+        vol_w_info = info.split()
+        if len(vol_w_info)>3:
+            if vol_w_info[2] == 'O' and "USER" in vol_w_info[3] and vol_w_info[4] == "PRIV/RSDNT":
+                ds_on_vol = hosts.all.shell(cmd=f"vtocls {vol_w_info[3]}")
+                for ds in ds_on_vol.contacted.values():
+                    datasets = str(ds.get("stdout")).split("\n")
+                    largest = 5
```

Comment:
I don't like this way of sorting, seems unnecessarily complex and not easy to understand. Check https://docs.python.org/3/howto/sorting.html#key-functions 

Why not create a tuple or object with the data set count and the [vol_w_info[3], vol_w_info[0]] like 


(2, [vol_w_info[3], vol_w_info[0]])

and then insert those into priv_online then sort with `priv_online.sort(key=lambda volume: volume[0])`
and only return the volume info as a list with `return [volume[1] for volume in priv_volume]`